netbox/netbox/utils.py

```diff
@@ -24,3 +25,14 @@ def _wrapper(cls):
         return cls
 
     return _wrapper
+
+
+def register_system_job():
+    """
+    Decorator for registering a `JobRunner` class as system background job.
+    """
+    def _wrapper(cls):
+        registry['system_jobs'][cls.name] = cls
```

Comment:
We probably want to namespace these by app or Python path. Otherwise, we risk a name collision among jobs from different plugins.